default/AI/ColonisationAI.py

```diff
@@ -103,6 +105,16 @@ def rate_planetary_piloting(pid):
         return 0.0
     return rate_piloting_tag(this_spec.tags)
 
+def calc_supply_tech_range():
+    supply_range = 0
+    for tech in AIDependencies.supply_range_techs:
+        if tech_is_complete(tech):
+            supply_distance += AIDependencies.supply_range_techs[tech]
```

Comment:
That is a typo, which I suppose I hadn't noticed since I'm actually using get_supply_tech_range() instead of this function.  (I think I wrote and swapped to that while in the middle of writing this one, but forgot to delete this one when done with the other).  I'll delete this function and probably change the other to use a summed generator expression and also use your turn_cache suggestion.